impl/src/main/java/org/ehcache/internal/concurrent/ConcurrentHashMap.java

```diff
@@ -6300,12 +6302,12 @@ public final void compute() {
         }
     }
 
-    public Set<Entry<K, V>> getRandomValues(Random rndm, int size, Predicate<Entry<K, V>> veto) {
-        Set<Entry<K, V>> sampled = new HashSet<Entry<K, V>>(size);
+    public List<Entry<K, V>> getRandomValues(Random rndm, int size, Predicate<Entry<K, V>> veto) {
```

Comment:
Are you suggesting to directly return the Node instances? Can't these change under your feet, making you potentially evict a candidate whose value has changed since you chose it?